iocage/lib/Jail.py

```diff
@@ -0,0 +1,500 @@
+
+    def set_routes(self):
+
+        defaultrouter = self.config.defaultrouter
+        defaultrouter6 = self.config.defaultrouter6
+
+        if not defaultrouter or defaultrouter6:
+            self.logger.spam("no static routes configured")
+            return
+
+        if defaultrouter:
+            self.logger.verbose(
+                f"setting default IPv4 gateway to {defaultrouter}",
+                jail=self
+            )
+            self._set_route(defaultrouter)
+
+        if defaultrouter6:
+            self._set_route(defaultrouter6, ipv6=True)
+
+    def _set_route(self, gateway, ipv6=False):
+
+        ip_version = 4 + 2 * (ipv6 is True)
+
+        self.logger.verbose(
+            f"setting default IPv{ip_version} gateway to {gateway}",
+            jail=self
+        )
+
+        command = [
+            "/sbin/route",
+            "add"
+        ] + (["-6"] if (ipv6 is True) else []) + [
+            "default",
+            gateway
+        ]
+
+        self.exec(command)
+
+    def require_jail_not_existing(self):
+        if self.exists:
+            msg = f"Jail {self.humanreadable_name} already exists"
+            self.logger.error(msg)
+            raise Exception(msg)
+
+    def require_jail_existing(self):
+        if not self.exists:
+            msg = f"Jail {self.humanreadable_name} does not exist"
+            self.logger.error(msg)
+            raise Exception(msg)
+
+    def require_jail_stopped(self):
+        if self.running:
+            msg = f"Jail {self.humanreadable_name} is already running"
+            self.logger.error(msg)
+            raise Exception(msg)
+
+    def require_jail_running(self):
+        if not self.running:
+            msg = f"Jail {self.humanreadable_name} is not running"
+            self.logger.error(msg)
+            raise Exception(msg)
+
+    def update_jail_state(self):
+        try:
+            stdout = subprocess.check_output([
+                "/usr/sbin/jls",
+                "-j",
+                self.identifier,
+                "-v",
+                "-h"
+            ], shell=False, stderr=subprocess.DEVNULL)
+            output = stdout.decode("utf-8").strip()
+
+            keys, values = [x.split(" ") for x in output.split("\n")]
+            self.jail_state = dict(zip(keys, values))
+
+        except:
+            self.jail_state = None
+
+    def _resolve_uuid(self, text):
+        jails_dataset = self.host.datasets.jails
+        for dataset in list(jails_dataset.children):
+            dataset_uuid = dataset.name[(len(jails_dataset.name) + 1):]
+            if text in dataset_uuid:
+                self.logger.debug(f"Resolved {text} to uuid {dataset_uuid}")
+                return dataset_uuid
+
+        raise Exception(f"No jail matching {text} was found")
+
+    def _get_name(self):
+        return self.humanreadable_name
+
+    def _get_humanreadable_name(self):
+
+        try:
+            return self.config.name
+        except:
```

Comment:
how can this return possibly generate an error? And if it does we should rightfully bubble that error and expose it instead of gobbling it up like this.

I see that you have a generic raise below but I fear that is not specific in locating the issue.